### PR TITLE
steal_time: Add new case to test steal time

### PR DESCRIPTION
**Pull request by huangyum**
ID: 1891722
Signed-off-by: Yumei Huang <yuhuang@redhat.com>

### CODE REVIEW

**Comment by huangyum** [comment]
(1/1) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.steal_time: PASS (267.56 s)

@yanan-fu Would you pls help review, thanks.

File: qemu/tests/steal_time.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+            stat_val.append(int(val))
+        return stat_val
+
+    stress_args = params["stress_args"]
+    stress_tests = []
+    sessions = []
+    vms = env.get_all_vms()
+
+    error_context.context("Run stress in both guests", logging.info)
+    for vm in vms:
```
Comment (yanan-fu) [inline qemu/tests/steal_time.py:39]: This design can not ensure the two vms run at the same time ?

File: qemu/tests/cfg/steal_time.cfg [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    vms += " vm0"
+    image_snapshot = yes
+    qemu_command_prefix = "taskset -c 1 "
+    smp = 1
+    vcpu_maxcpus = 1
+    vcpu_sockets = 1
+    vcpu_threads = 1
+    stress_args = --cpu 1 --vm 2 --vm-bytes 1024 
+    top_cmd = top -n 1 | grep Cpu
+    stat_cmd = cat /proc/stat | head -1
```
Comment (yanan-fu) [inline qemu/tests/cfg/steal_time.cfg:16]: Use the sum of all cpus ? No the one be used in taskset ?

Comment (huangyum) [reply qemu/tests/cfg/steal_time.cfg:16]: It will run inside guest, should have nothing to do with taskset.

File: qemu/tests/steal_time.py (hunk 1 above)
Comment (huangyum) [reply qemu/tests/steal_time.py:39]: As long as they are both running and stressed before check steal time, then it's fine.

**Comment by huangyum** [comment]
@yanan-fu Any more comments ?

File: qemu/tests/steal_time.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Steal time test:
+    1) Boot two guests and bind to one same cpu
+    2) Run stress inside both guests
+    3) Check steal time in top inside guests
+    4) Check steal time in /proc/stat in guest
```
Comment (yanan-fu) [inline qemu/tests/steal_time.py:16]: No need to check the qemu-kvm cpu usage in host ?

File: qemu/tests/steal_time.py [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+        stat_val_pre = get_stat_val()
+        logging.info("Steal time value in /proc/stat is %s", stat_val_pre)
+        time.sleep(60)
+        stat_val_post = get_stat_val()
+        logging.info("After 60s, steal time value in /proc/stat is %s",
+                     stat_val_post)
+
+        delta = list(map(lambda x, y: y - x, stat_val_pre, stat_val_post))
+        if abs(delta[0] - delta[1]) > delta[1]*0.1:
```
Comment (yanan-fu) [inline qemu/tests/steal_time.py:67]: I have a bit concern about the threshold `delta[1]*0.1` but no more good idea now. how about use the average `(delta[0] + delta[1]/2)*0.1` ?

Comment (huangyum) [reply qemu/tests/steal_time.py:79]: Actually the difference between delta[0] and delta[1] is very small, much smaller than 10% of both. So I think it's fine. Thanks.

File: qemu/tests/steal_time.py (hunk 3 above)
Comment (huangyum) [reply qemu/tests/steal_time.py]: Yes, I may have missed this step, updated, pls help review again, thanks.

**Comment by huangyum** [comment]
Latest test resutls:
100-repeat100.Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.steal_time' Finshed 2020-11-09 14:35:06 2020-11-09 14:40:07 PASS 301.589356

File: qemu/tests/steal_time.py (hunk 4 above)
Comment (yanan-fu) [reply qemu/tests/steal_time.py:79]: 'difference is small' is the normal case i think.  I just thought use a value that may be wrong as threshold is not very suitable, so suggest average in the pre comment.
Does fixed threshold work with this situation ?

**Review by yanan-fu — APPROVED** [review]
Acked-by: Yanan Fu <yfu@redhat.com>

**Pull request merged**